lib/src/DynamicProto.ts

```diff
@@ -229,44 +250,96 @@ function _getBaseFuncs(classProto:any, thisTarget:any, instFuncs:any): any {
+                    canAddInst = (protoFunc === currentDynProtoProxy);
+                    break;
                 }
 
-                // Avoid stack overflow from recursive calling the same function
-                _throwTypeError("Missing [" + funcName + "] " + strFunction);
+                // We need to find all possible initial functions to ensure that we don't bypass a valid override function
+                visited.push(objProto);
+                objProto = _getObjProto(objProto);
             }
 
-            let protoFunc = proto[funcName];
+            try {
+                if (canAddInst) {
+                    // This instance doesn't have an instance func and the class hierarchy does have a higher level prototype version
+                    // so it's safe to directly assign for any subsequent calls (for better performance)
+                    target[funcName] = instFunc;
+                }
 
-            // Check that the prototype function is not a self reference -- try to avoid stack overflow!
-            if (protoFunc === dynProtoProxy) {
-                // It is so lookup the base prototype
-                protoFunc = _getObjProto(proto)[funcName];
+                // Block further attempts to set the instance function for any
+                instFunc[DynInstChkTag] = 1;
+            } catch (e) {
+                // Don't crash if the object is readonly or the runtime doesn't allow changing this
+                // And set a flag so we don't try again for any function
+                instFuncTable[DynAllowInstChkTag] = false;
             }
+        }
+    }
 
-            if (!_isFunction(protoFunc)) {
-                _throwTypeError("[" + funcName + "] is not a " + strFunction);
-            }
+    return instFunc;
+}
+
+function _getProtoFunc(funcName: string, proto: any, currentDynProtoProxy: any) {
+    let protoFunc = proto[funcName];
+
+    // Check that the prototype function is not a self reference -- try to avoid stack overflow!
+    if (protoFunc === currentDynProtoProxy) {
+        // It is so lookup the base prototype
+        protoFunc = _getObjProto(proto)[funcName];
+    }
+
+    if (!_isFunction(protoFunc)) {
+        _throwTypeError("[" + funcName + "] is not a " + strFunction);
+    }
 
-            return protoFunc.apply(_this, arguments);
+    return protoFunc;
+}
+
+/**
+ * Add the required dynamic prototype methods to the the class prototype
+ * @param proto - The class prototype
+ * @param className - The instance classname 
+ * @param target - The target instance
+ * @param baseInstFuncs - The base instance functions
+ * @param setInstanceFunc - Flag to allow prototype function to reset the instance function if one does not exist
+ * @ignore
+ */
+function _populatePrototype(proto:any, className:string, target:any, baseInstFuncs:any, setInstanceFunc:boolean) {
+    function _createDynamicPrototype(proto:any, funcName:string) {
+        var dynProtoProxy = function() {
```

Comment:
Simplifying the dynamicProto proxy so that when you debug through this function is now only 2 lines long to get to the real function.